state-chain/chains/src/dot.rs

```diff
@@ -982,6 +987,8 @@ pub struct PolkadotReplayProtection {
 	pub nonce: PolkadotIndex,
 }
 
+pub type PolkadotCheckMetadataHash = Option<()>;
```

Comment:
I didn't suggest not using this, as the `Mode` enum they use is like [this](https://docs.rs/frame-metadata-hash-extension/0.3.0/src/frame_metadata_hash_extension/lib.rs.html#53):

```rust
/// The mode of [`CheckMetadataHash`].
#[derive(Decode, Encode, PartialEq, Debug, TypeInfo, Clone, Copy, Eq)]
enum Mode {
	Disabled,
	Enabled,
}
```

And is on reason to believe the Option encoding would be the same as that.

Reply:
My understanding is that the Mode is what is used in the runtime (ie. in Polkadot) to tell the runtime whether or not it supports this. 

The associated `AdditionalSigned` is what is actually signed over, which is an `Option<[u8; 32>`. See [here](https://github.com/paritytech/polkadot-sdk/blob/8dbe4ee80734bba6644c7e5f879a363ce7c0a19f/substrate/frame/metadata-hash-extension/src/lib.rs). 

What's a bit crazy is that even if the runtime *doesn't* support it, you need to add `None` to the signed payload...

Reply:
As discussed - you were right. I've updated it.